src/browser-lib/cookie-consent/CookieConsentProvider.tsx

```diff
@@ -3,83 +3,127 @@
+} from "@oaknational/oak-components";
 
-export type CookieConsents = Record<CookiePolicyName, CookieConsent>;
-export type HasConsentedTo = (serviceType: ServiceType) => CookieConsentState;
-export type HasConsentedToPolicy = (
-  policyName: CookiePolicyName,
-) => CookieConsentState;
+import { consentClient } from "./consentClient";
 
 export type CookieConsentContext = {
-  // makes consent manager modal visible
   showConsentManager: () => void;
-  // whether the user has granted consent to the latest version of a partular policy
-  hasConsentedTo: HasConsentedTo;
-  // consent by policyName
-  hasConsentedToPolicy: HasConsentedToPolicy;
+  getConsentState: (
+    policyName: "strictly-necessary" | "embedded-content" | "statistics",
+  ) => "granted" | "denied" | "pending";
 };
 
 export const cookieConsentContext = createContext<CookieConsentContext | null>(
   null,
 );
 
 export const useCookieConsent = () => {
-  const cookieConsentsContext = useContext(cookieConsentContext);
-  if (!cookieConsentsContext) {
+  const context = useContext(cookieConsentContext);
+
+  if (!context) {
     throw new Error(
       "useCookieConsent() called outside of cookieConsentContext provider",
     );
   }
-  return cookieConsentsContext;
-};
 
-type CookieConsentProviderProps = {
-  children?: React.ReactNode;
-  __testMockValue?: CookieConsentContext;
+  return context;
 };
-const CookieConsentProvider: FC<CookieConsentProviderProps> = (props) => {
-  const { children } = props;
-  const consents = useConfirmicConsents();
 
-  const showConsentManager = () => {
-    // eslint-disable-next-line
-    // @ts-ignore
-    window.Metomic("ConsentManager:show");
-  };
+const CookieConsentContextProvider = (props: PropsWithChildren) => {
+  const { state, getConsent: getConsentState } = useOakConsent();
+  const { showBanner, openSettings: showConsentManager } = useCookieConsentUI();
 
-  const hasConsentedTo = (serviceType: ServiceType) => {
-    const policyName = servicePolicyMap[serviceType];
+  useEffect(() => {
+    if (state.requiresInteraction) {
+      showBanner();
+    }
+  }, [state.requiresInteraction, showBanner]);
 
-    return consents[policyName].state;
-  };
+  return (
+    <cookieConsentContext.Provider
+      {...props}
+      value={{ getConsentState, showConsentManager }}
+    />
+  );
+};
 
-  const hasConsentedToPolicy = (policyName: CookiePolicyName) => {
-    return consents[policyName].state;
-  };
+const CookieConsentUIProvider = ({ children }: PropsWithChildren) => {
+  const { state, logConsents } = useOakConsent();
+  const policies = state.policyConsents.map((policyConsent) => ({
+    id: policyConsent.policyId,
+    label: policyConsent.policyLabel,
+    description: policyConsent.policyDescription,
+    strictlyNecessary: policyConsent.isStrictlyNecessary,
+    parties: policyConsent.policyParties.map((party) => ({
+      name: party.name,
+      policyURL: party.url,
+    })),
+  }));
```

Comment:
I plan to open a PR in oak-components to rename these properties to match oak-consent-client making this mapping redundant.